proto/directions.proto

```diff
@@ -0,0 +1,94 @@
+package valhalla;
+
+message Waypoint {
+  optional float lat = 1;
+  optional float lon = 2;
+  optional string type = 3;
+  optional uint32 bearing = 4;
+  optional string name = 5;
+  optional string street = 6;
+  optional string side_of_street = 7;
+}
+
+message BoundingBox {
+  optional float min_lat = 1;
+  optional float min_lon = 2;
+  optional float max_lat = 3;
+  optional float max_lon = 4;
+}
+
+message StreetName {
+  optional string text = 1;  // Text of street name, examples: MD 43 East, White Marsh Boulevard 
+  optional bool is_ref = 2;  // true if street name is a ref, examples: true, false
+}
+
+message Step {
+  optional float distance = 1;
+  optional uint32 duration = 2;
+  optional uint32 geometry_index_begin = 3;      // begin geometry inclusive index
+  optional uint32 geometry_index_end = 4;        // end geometry inclusive index
+  optional uint32 maneuver_index_begin = 5;      // the index of the maneuver at the start of the step
+  optional uint32 maneuver_index_end = 6;        // the index of the maneuver at the end of the step
+  repeated StreetName street_names = 7;          // consistent street names for the entire step
+  optional string travel_mode = 8;               // travel mode 
+  optional string travel_mode_type = 9;          // travel type
+  optional string driving_side = 10;             // legal driving side for this step
+}
+  
+message SignElement {
+  optional string text = 1;               // The actual sign element text, example: I 95 North
+  optional bool is_ref = 2;               // true if sign element is a ref, example: true  
+  optional uint32 consecutive_count = 3;  // The frequency of this sign element within a set a consecutive signs
+}
+
+message Sign {
+  repeated SignElement exit_number_elements = 1;  // The list of exit numbers, example: 67B
+  repeated SignElement exit_branch_elements = 2;  // The list of exit branch street names, example: I 95 North
+  repeated SignElement exit_toward_elements = 3;  // The list of exit toward locations, example: New York
+  repeated SignElement exit_name_elements   = 4;  // The list of exit names - not used much in US, example: Gettysburg Pike
+}
+
+message Lane {
+  optional bool valid = 1;
+  repeated string indications = 2;
+}
+
+message Maneuver {
+  optional string type = 1;
+  optional uint32 geometry_index = 2;
+  optional uint32 bearing_before = 3;   // incoming bearing at transition 
+  optional uint32 bearing_after = 4; 
+  repeated StreetName street_names = 5; // street names at the transition
+  optional Sign sign = 6;
+  optional Lane lane = 7;
+  optional bool obvious = 8;            // true if transition is obvious, might not need announced
+  optional bool verbal_multi_cue = 9;   // true if the pre transition also announces the next transition alert
```

Comment:
depending on how we wind up doing voice instructions, this type of decision may happen on the nav-native side.